src/App.tsx

```diff
@@ -148,6 +165,58 @@ function App() {
     });
   }, []);
 
+  useEffect(() => {
+    async function getPipelineVersions(pipelineURI: FieldInputOption) {
+      try {
+        const response: AxiosResponse<RetrievedPipelineVersions> = await axios.get(
+          `${baseAPIURL}pipelines/${pipelineURI.id}/versions`
+        );
+        if (response.data.nodes_response_status === 'fail') {
+          enqueueSnackbar(`Failed to retrieve ${pipelineURI.label} versions`, {
+            variant: 'error',
+            action,
+          });
+        } else {
+          // If any errors occurred, report them
+          response.data.errors.forEach((error) => {
+            enqueueSnackbar(
+              `Failed to retrieve ${pipelineURI.label} versions from ${error.node_name}`,
+              {
+                variant: 'warning',
+                action,
+              }
+            );
+          });
+          // If the results are empty, report that
+          if (Object.keys(response.data.responses[pipelineURI.id]).length === 0) {
+            enqueueSnackbar(`No ${pipelineURI.label} versions were available`, {
+              variant: 'info',
+              action,
+            });
+          }
+        }
+        return response.data.responses[pipelineURI.id];
+      } catch {
+        return [];
+      }
+    }
+    // Get pipeline versions if
+    // 1. A pipeline has been selected (this implementation only works for single value for pipeline name field)
+    // 2. This is the first time its being selected (i.e., we haven't retrieved pipeline versions before)
+    if (
+      pipelineName !== null &&
+      !Array.isArray(pipelineName) &&
```

Comment:
Is this line checking whether I have selected multiple pipelines? Can I even do that right now?

Reply:
Kinda, it's way to get around the `FieldInputOption []` which is part of the `FieldInput` and prevent the type error.